### PR TITLE
Updated Pedal Arduino Code

### PR DESCRIPTION
**Pull request by rydzynskip**
Updated code for only two potentiometers and a brake switch.

### CODE REVIEW

**Review by mccauleyma — CHANGES_REQUESTED** [review]
Not to be a pain in the ass. But should we replace "gas" with accelerator/accel?

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 1]
```diff
@@ -44,27 +43,17 @@ void setup()
...
-    unsigned char byteReading = averageReading/4;
-    unsigned char valueMessage[8] = {byteReading, 0, 0, 0, 0, 0, 0, 0};
-    
-    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message of the average value of the two brake potentiometers
-  }
-  else{
-    CAN.sendMsgBuf(myCAN_brake_broke, 0, 8, errorMessage); //send message that the two potentiometer values are too far apart
-  }
-
   if(abs(gasPin1Val-gasPin2Val) <= 1024*0.1){
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:68]:
```suggestion
   if (abs(gasPin1Val - gasPin2Val) <= 1024 * 0.1) {
```

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 2]
```diff
@@ -44,27 +43,17 @@ void setup()
...
   gasPin2Val = analogRead(gasPin2);
 
-  if(abs(brakePin1Val-brakePin2Val) <= 1024*0.1){
-    int averageReading = (brakePin1Val + brakePin2Val)/2;
-    unsigned char byteReading = averageReading/4;
-    unsigned char valueMessage[8] = {byteReading, 0, 0, 0, 0, 0, 0, 0};
-    
-    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message of the average value of the two brake potentiometers
-  }
-  else{
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:64]:
```suggestion
  } else {
```
Collapse into one line, styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 3]
```diff
@@ -44,27 +43,17 @@ void setup()
...
   gasPin1Val = analogRead(gasPin1); 
   gasPin2Val = analogRead(gasPin2);
 
-  if(abs(brakePin1Val-brakePin2Val) <= 1024*0.1){
-    int averageReading = (brakePin1Val + brakePin2Val)/2;
-    unsigned char byteReading = averageReading/4;
-    unsigned char valueMessage[8] = {byteReading, 0, 0, 0, 0, 0, 0, 0};
-    
-    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message of the average value of the two brake potentiometers
-  }
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:63]:
```suggestion
```

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 4]
```diff
@@ -75,7 +64,20 @@ void loop() {
   else{
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:75]: Styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 5]
```diff
@@ -75,7 +64,20 @@ void loop() {
...
     CAN.sendMsgBuf(myCAN_gas_broke, 0, 8, errorMessage); //send message that the two potentiometer values are too far apart
   }
-  
+
+  // reading switch values for brake pedal, comparing
+  // them and then sending a message to the CAN
+  brakePin1Val = digitalRead(brakePin1);
+  brakePin2Val = digitalRead(brakePin2);
+
+  if(brakePin1Val == LOW && brakePin2Val == LOW){
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:73]: Add parentheses for clarity and readability

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 6]
```diff
@@ -75,7 +64,20 @@ void loop() {
...
+    unsigned char valueMessage[8] = {1, 0, 0, 0, 0, 0, 0, 0};
+    
+    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message that the switches are turned on
+  }
+  else if(brakePin1Val != brakePin2Val){
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:78]: Styling, collapse to one line

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 7]
```diff
@@ -75,7 +64,20 @@ void loop() {
...
+    CAN.sendMsgBuf(myCAN_brake_broke, 0, 8, errorMessage); //send message that the two switch values are not equal
+  }
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:80]: Remove erroneous blank lines

**Review by jamescd18 — CHANGES_REQUESTED** [review]
This is really awesome to see! A lot of styling changes plus JavaDoc-styling comments would really help with the readability, especially because we're working in a team.

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+int brakePin1Val;
+int brakePin2Val;
+ 
+// CAN stuff
+const int myCAN_accel = 0x01;       // device's CAN id
+const int myCAN_accel_broke = 0x02; // device error ID
+const int myCAN_brake = 0x03;       // device's CAN id
+const int myCAN_brake_broke = 0x04; // device error ID
+
+const int spiCSPin = 10; //base CAN pin
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:31]: Constants should be named with all caps and underscores.

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+unsigned char errorMessage[8] = {0, 255, 0, 0, 0, 0, 0, 0}; // output message for when the potentiometers are too far apart
+// this message can be changed, but for now has value in second index as no "good" message will have a value there
+
+MCP_CAN CAN(spiCSPin); // setup can device
+
+/**
+ * Start CAN and initialize digital pins
+ */
+void setup()
+{
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:41]:
```suggestion
void setup() {
```
Styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  Serial.begin(115200); // baud rate for CAN
+
+  while (CAN_OK != CAN.begin(CAN_500KBPS, MCP_8MHz)) // specify 8MHz crystal
+  {
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:45]:
```suggestion
  while (CAN_OK != CAN.begin(CAN_500KBPS, MCP_8MHz)) { // specify 8MHz crystal
```
Styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+ */
+void readSwitches() {
+  brakePin1Val = digitalRead(brakePin1);
+  brakePin2Val = digitalRead(brakePin2);
+
+  if ((brakePin1Val == LOW) && (brakePin2Val == LOW)) {
+    unsigned char valueMessage[8] = {1, 0, 0, 0, 0, 0, 0, 0};
+    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message that the switches are turned on
+  } else if (brakePin1Val != brakePin2Val) {
+    CAN.sendMsgBuf(myCAN_brake_broke, 0, 8, errorMessage); //send message that the two switch values are not equal
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:94]:
```suggestion
    CAN.sendMsgBuf(myCAN_brake_broke, 0, 8, errorMessage); // send message that the two switch values are not equal
```
Styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+ * Read the values of the two switches
+ * Compare them to make sure the values agree and send value to CAN
+ */
+void readSwitches() {
+  brakePin1Val = digitalRead(brakePin1);
+  brakePin2Val = digitalRead(brakePin2);
+
+  if ((brakePin1Val == LOW) && (brakePin2Val == LOW)) {
+    unsigned char valueMessage[8] = {1, 0, 0, 0, 0, 0, 0, 0};
+    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); //send message that the switches are turned on
```
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:92]:
```suggestion
    CAN.sendMsgBuf(myCAN_brake, 0, 8, valueMessage); // send message that the switches are turned on
```
Styling

File: ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino (hunk 8 above)
Comment (jamescd18) [inline ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino:31]:
```suggestion
const int spiCSPin = 10; // base CAN pin
```
Styling

**Review by jamescd18 — CHANGES_REQUESTED** [review]
I absolutely love the additional commenting, function documentation, and code separation! It makes it 10x more understandable and readable for those that have never seen the code before. Two things: just a few styling edits here and there, and then also is there a reason for nesting the file another directory level deeper? I'm a little confused why it went from `ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers.ino` to `ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino`. Other than those, it's really awesome to see the improvements. Keep it up!

**Comment by mccauleyma** [comment]
> I absolutely love the additional commenting, function documentation, and code separation! It makes it 10x more understandable and readable for those that have never seen the code before. Two things: just a few styling edits here and there, and then also is there a reason for nesting the file another directory level deeper? I'm a little confused why it went from `ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers.ino` to `ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino`. Other than those, it's really awesome to see the improvements. Keep it up!

@jamescd18 The nested folders is one of the weird requirements of Arduino's IDE and .ino files. The IDE requires nested folders before allowing you to open the file. It's annoying but ¯\_(ツ)_/¯

**Review by mccauleyma — APPROVED** [review]

**Comment by jamescd18** [comment]
> @jamescd18 The nested folders is one of the weird requirements of Arduino's IDE and .ino files. The IDE requires nested folders before allowing you to open the file. It's annoying but ¯_(ツ)_/¯

That would explain it. Can it be adjusted to `ArduinoCode/Pedal_Potentiometers/Pedal_Potentiometers.ino`instead of `ArduinoCode/Pedal Potentiometers/Pedal_Potentiometers/Pedal_Potentiometers.ino`?

**Comment by mccauleyma** [comment]
Yep. That would work. Only needs the direct parent folder. Ignores anything above that.

**Review by jamescd18 — APPROVED** [review]
Great job! I love all the improvements.

**Pull request merged**